### PR TITLE
Deprecating old github/azure-iot-sdk-c Arduino libraries

### PR DESCRIPTION
**Pull request by ewertons**
These libraries are being removed and customers are recommended
to use the new azure-sdk-for-c Arduino library. Deprecation
warnings were added 6 months ago to each of the libraries
unlisted in this change.

This change has been approved by the Azure IoT C SDK Team.

### CODE REVIEW

**Review by danewalton — APPROVED** [review]

**Review by ericwol-msft — APPROVED** [review]

**Comment by per1234** [comment]
Hello @ewertons. I am confirming your request.

Unlike submissions, this type of request is still handled manually. The Library Manager index maintainer occasionally comes through to clear all the outstanding requests. They will merge this pull request when it is done. You will get a notification from GitHub at that time.

**Comment by umbynos** [comment]
Done in https://github.com/arduino/library-registry/pull/1640#issuecomment-1204120160

**Review by umbynos — APPROVED** [review]

**Pull request merged**
